### PR TITLE
cc3 + rm

### PR DESCRIPTION
**Pull request by baxance**
- [x] Top-level README “Table of Contents” is updated
 - [x] Feature tasks for this challenge are completed
 - [x] Unit tests written and passing
     - [x] “Happy Path” - Expected outcome
     - [x] Expected failure
     - [x] Edge Case (if applicable/obvious)
 - [x] README for this challenge is complete
     - [x] Summary, Description, Approach & Efficiency, Solution
     - [x] Link to code
     - [x] Picture of whiteboard

### CODE REVIEW

File: java/challenges/lib/src/test/java/challenges/BinarySearchTest.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
+package challenges;
+
+import org.junit.Test;
+import static org.junit.Assert.*;
+
+public class BinarySearchTest {
+
+  @Test
```
Comment (Mattpet26) [inline java/challenges/lib/src/test/java/challenges/BinarySearchTest.java:8]: Good work on the test, I'd suggest writing a few more in the future. What if the array was empty? What if the key didn't exist in the array?
